### PR TITLE
Drop Python 2 support

### PR DESCRIPTION
**Pull request by cdeil**
This PR drops Python 2 support, as discussed in #111 .

Work in progress, do not merge. (I just want to see if CI builds still pass)

### CODE REVIEW

**Comment by cdeil** [comment]
@astrofrog or @lpsinger or @bsipocz  or anyone - please review.

The CI fails are unrelated: what's this numpy 1.10 install issue in https://travis-ci.org/astropy/astropy-healpix/jobs/489434822#L964 about? Maybe just drop that CI build?

**Comment by lpsinger** [comment]
LGTM

File: .travis.yml [hunk 1]
```diff
@@ -95,11 +91,11 @@ matrix:
         # time.
 
         - os: linux
-          env: PYTHON_VERSION=2.7 NUMPY_VERSION=1.12
+          env: PYTHON_VERSION=3.6 NUMPY_VERSION=1.12
```
Comment (bsipocz) [inline .travis.yml]: what about python 3.7? I suppose that's OK to do it in a separate PR.

Comment (cdeil) [reply .travis.yml]: You think there's value in adding Python 3.7 builds? Or should some be replaced?

In my experience passing on Python 3.6 and failing on Python 3.7 is super rare, not worth the 🔌 , better to save a few 🌴 🎋 🎄 🌳 🌲 s. But I don't care much, just le me know what you suggest or put it directly.

**Comment by bsipocz** [comment]
For the numpy 1.10 build you need to use `ASTROPY_VERSION=lts` as the stable version is not compatible with it.

Also, just for some good measures, please remove all but the astropy channel from `CONDA_CHANNELS`, mixing conda-forge with defaults can lead to all kind of issues.

**Comment by bsipocz** [comment]
(E.g. point no2 was about this CI version issue in here: https://github.com/astropy/astropy-healpix/issues/114)

File: .travis.yml (hunk 1 above)
Comment (bsipocz) [reply .travis.yml]: I suggest to replace. In other packages we scatter versions around to cover the combinations. E.g. no need to test different numpy versions with the same python, change both to you can cover more (and can rely on the assumption that all upstream packages are testing against the remaining combinations, that is a sensible assumption).

Comment (bsipocz) [reply .travis.yml]: Also I would consider dropping a few numpy versions from support, 1.10 is 3 years old already. Or if you don't want it then test the oldest one you support, but then cover the newer ones rather than test all the old ones.

**Comment by bsipocz** [comment]
@cdeil - I'm checking the list of packages still importing `six` from astropy, and thus run into astropy-healpix. 

Could we go ahead with this PR? After a rebase I suspect that the failure against the dev version will go away (and hopefully no new ones will pop up).

**Comment by cdeil** [comment]
conda install fail with Python 3.5 on Windows: https://ci.appveyor.com/project/Astropy/astropy-healpix/builds/25712558
Changing to 3.6 for now.

@astrofrog - do you feel Python 3.5 support is really important here, or would you be OK if we drop it and go to 3.6+ to make our lives a little bit easier?

**Comment by astrofrog** [comment]
I think we can just stay in sync with astropy core, which means supporting only Python 3.6+ now

**Comment by cdeil** [comment]
Great! I'll change to `>=3.6` in `setup.py` and `.travis.yml` and maybe do a little bit more cleanup using 3.6 features then.

**Comment by hugovk** [comment]
Here's a tool that can upgrade the syntax for Python 3.6, f-strings and so on:

```sh
pip install -U pyupgrade
pyupgrade `find . -name "*.py" -type f` --py36-plus
```

**Comment by cdeil** [comment]
@bsipocz @astrofrog @lpsinger - Anyone has time for a final review here?

**Comment by cdeil** [comment]
> We should probably add a Python 3.7 config for testing

Done in d2fff67
